File: input/java/Person02.java
import java.util.Calendar;
import java.util.GregorianCalendar;
import java.util.Random;

public class Person02 {
    private String firstName;
    private String lastName;
    private Calendar birthday;
    private int favouriteNumber;

    public Person02()
    {
        this.firstName = "";
        this.lastName = "";
        this.birthday = new GregorianCalendar();

        Random randomObj = new Random();
        this.favouriteNumber = randomObj.nextInt(42);
    }
    public Person02(String firstName, String lastName)
    {
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthday = new GregorianCalendar();

        Random randomObj = new Random();
        this.favouriteNumber = randomObj.nextInt(42);
    }
    public Person02(String firstName, String lastName, Calendar birthday)
    {
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthday = birthday;

        Random randomObj = new Random();
        this.favouriteNumber = randomObj.nextInt(42);
    }

    public String fullName()
    {
        return firstName + " " + lastName;
    }

    public int age(Calendar today)
    {
        return today.get(Calendar.YEAR) - birthday.get(Calendar.YEAR);
    }

    public int favNum() {
        return favouriteNumber;
    }
}

File: input/kotlin/Person02.kt
import java.util.*

class Person02 {
    private var firstName: String
    private var lastName: String
    private var birthday: Calendar
    private var favouriteNumber: Int

    constructor() {
        firstName = ""
        lastName = ""
        birthday = GregorianCalendar()
        val randomObj = Random()
        favouriteNumber = randomObj.nextInt(42)
    }

    constructor(firstName: String, lastName: String) {
        this.firstName = firstName
        this.lastName = lastName
        birthday = GregorianCalendar()
        val randomObj = Random()
        favouriteNumber = randomObj.nextInt(42)
    }

    constructor(firstName: String, lastName: String, birthday: Calendar) {
        this.firstName = firstName
        this.lastName = lastName
        this.birthday = birthday
        val randomObj = Random()
        favouriteNumber = randomObj.nextInt(42)
    }

    fun fullName(): String {
        return "$firstName $lastName"
    }

    fun age(today: Calendar): Int {
        return today[Calendar.YEAR] - birthday[Calendar.YEAR]
    }

    fun favNum(): Int {
        return favouriteNumber
    }
}